RecoTracker/LSTCore/src/alpaka/Triplet.h

```diff
@@ -848,15 +849,28 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
         int category_number = getCategoryNumber(module_layers, module_subdets, module_rings);
         int eta_number = getEtaBin(module_eta);
 
-        int occupancy = 0;
-        if (category_number != -1 && eta_number != -1) {
-          occupancy = occupancy_matrix[category_number][eta_number];
+        int dynamic_count = 0;
+        // How many segments are in module i?
+        int nSegments_i = segmentsOccupancy.nSegments()[i];
+        int firstSegmentIdx = ranges.segmentRanges()[i][0];
+        // Loop over all segments that live in module i
+        for (int s = 0; s < nSegments_i; ++s) {
+          int segIndex = firstSegmentIdx + s;
+          uint16_t midModule = segments.outerLowerModuleIndices()[segIndex];
+          dynamic_count += segmentsOccupancy.nSegments()[midModule];
         }
+
 #ifdef WARNINGS
-        else {
+        if (category_number == -1 || eta_number == -1) {
           printf("Unhandled case in createTripletArrayRanges! Module index = %i\n", i);
         }
 #endif
+        // Get matrix-based cap
+        int matrix_cap =
+            (category_number != -1 && eta_number != -1) ? occupancy_matrix[category_number][eta_number] : dynamic_count;
+
+        // Cap occupancy at minimum of dynamic count and matrix value
+        int occupancy = alpaka::math::min(acc, dynamic_count, matrix_cap);
 
         ranges.tripletModuleOccupancy()[i] = occupancy;
         unsigned int nTotT = alpaka::atomicAdd(acc, &nTotalTriplets, occupancy, alpaka::hierarchy::Threads{});
```

Comment:
ah, I guess `unsigned int` in the `Segment.h` arises from here.
To be stricter, unsigned for nX is better, but then perhaps the inputs should be unsigned as well (`nTotalTriplets` and `occupancy` are both signed)

Reply:
I decided to leave all of left the signs the way they are now, since I think it would be better to do dedicated code cleanup like this in its own PR.